### PR TITLE
Add glm engine for `linear_reg()`

### LINKED ISSUE #623: glm engine for linear reg
... and make sure that we can pass options like `family = statmod::tweedie()`

### PR DESCRIPTION
**Pull request by juliasilge**
Closes #623 

``` r
library(parsnip)

linear_reg() %>% 
  set_engine("glm") %>%
  translate()
#> Linear Regression Model Specification (regression)
#> 
#> Computational engine: glm 
#> 
#> Model fit template:
#> stats::glm(formula = missing_arg(), data = missing_arg(), weights = missing_arg(), 
#>     family = stats::gaussian)
```

<sup>Created on 2021-12-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

**Comment by juliasilge** [comment]
We can pass in expressions to the `family` engine argument:

``` r
library(parsnip)

spec1 <- 
  linear_reg() %>% 
  set_engine("glm", family = statmod::tweedie) 

spec1 %>% translate()
#> Linear Regression Model Specification (regression)
#> 
#> Engine-Specific Arguments:
#>   family = statmod::tweedie
#> 
#> Computational engine: glm 
#> 
#> Model fit template:
#> stats::glm(formula = missing_arg(), data = missing_arg(), weights = missing_arg(), 
#>     family = statmod::tweedie)

spec1 %>% fit(breaks ~ ., data = warpbreaks)
#> parsnip model object
#> 
#> 
#> Call:  stats::glm(formula = breaks ~ ., family = ~statmod::tweedie, 
#>     data = data)
#> 
#> Coefficients:
#> (Intercept)        woolB     tensionM     tensionH  
#>      39.278       -5.778      -10.000      -14.722  
#> 
#> Degrees of Freedom: 53 Total (i.e. Null);  50 Residual
#> Null Deviance:       9233 
#> Residual Deviance: 6748  AIC: NA

spec2 <- 
  linear_reg() %>% 
  set_engine("glm", family = stats::poisson(link = "sqrt")) 

spec2 %>% translate()
#> Linear Regression Model Specification (regression)
#> 
#> Engine-Specific Arguments:
#>   family = stats::poisson(link = "sqrt")
#> 
#> Computational engine: glm 
#> 
#> Model fit template:
#> stats::glm(formula = missing_arg(), data = missing_arg(), weights = missing_arg(), 
#>     family = stats::poisson(link = "sqrt"))

spec2 %>% fit(breaks ~ ., data = warpbreaks)
#> parsnip model object
#> 
#> 
#> Call:  stats::glm(formula = breaks ~ ., family = ~stats::poisson(link = "sqrt"), 
#>     data = data)
#> 
#> Coefficients:
#> (Intercept)        woolB     tensionM     tensionH  
#>      6.2620      -0.5059      -0.8545      -1.3644  
#> 
#> Degrees of Freedom: 53 Total (i.e. Null);  50 Residual
#> Null Deviance:       297.4 
#> Residual Deviance: 212.7     AIC: 495.3
```

<sup>Created on 2021-12-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

File: man/rmd/linear_reg_glm.Rmd [hunk 1]
```diff
@@ -0,0 +1,29 @@
+```{r, child = "setup.Rmd", include = FALSE}
+```
+
+`r descr_models("linear_reg", "glm")`
+
+## Tuning Parameters
+
+This engine has no tuning parameters but you can set the `family` parameter (and `link`) as an engine argument. 
+
```
Comment (topepo) [inline man/rmd/linear_reg_glm.Rmd:9]: Maybe mention that you can pass `family` and `link` in without issues.

Comment (juliasilge) [reply man/rmd/linear_reg_glm.Rmd:9]: How do you mean specifically, different than what I added already?

Comment (topepo) [reply man/rmd/linear_reg_glm.Rmd:9]: 😬

**Review by EmilHvitfeldt — APPROVED** [review]

**Pull request merged**